example/javascriptlib/publishing/2-realistic/build.mill

```diff
@@ -0,0 +1,124 @@
+package build
+
+import mill._, javascriptlib._
+
+object foo extends TypeScriptModule {
+  def generatedSources = Task {
+    os.write(
+      Task.dest / "foo.generated.ts",
+      s"""export default class FooGen {
+         |  static value: number = 123
+         |}
+      """.stripMargin
+    )
+
+    Seq(PathRef(Task.dest))
+  }
+
+  object bar extends TypeScriptModule {
+    def generatedSources = Task {
+      os.write(
+        Task.dest / "bar.generated.ts",
+        s"""export default class Bar {
+           |  static value: number = 123
+           |}
+      """.stripMargin
+      )
+
+      Seq(PathRef(Task.dest))
+    }
+
+    def npmDeps = Seq("immutable@4.3.7")
+  }
+
+}
+
+class qux extends TypeScriptModule {
+  def moduleDeps = Seq(foo, foo.bar)
+
+  def generatedSources = Task {
+    os.write(
+      Task.dest / "qux.generated.ts",
+      s"""export default class QuxGen {
+         |  static value: number = 123
+         |}
+      """.stripMargin
+    )
+
+    Seq(PathRef(Task.dest))
+  }
+
+  object test extends TypeScriptTests with TestModule.Jest
+}
+
+object quxdev extends qux
```

Comment:
Do we still need `quxdev` if `quxpub` exists?

Reply:
I was trying to demonstrate in the example that is it possible to have a dev (avoid potentially expensive transpile, path resolution process & publish prep. for very large project) and prod version of qux  as its a common practice in js/ts development.

I have removed it from the example.

Reply:
no